polynomial/core.py

```diff
@@ -174,8 +174,8 @@ def __setattr__(self, name, new_value):
     def __getitem__(self, degree):
         """Get the coefficient of the term with the given degree."""
         if isinstance(degree, slice):
-            indices = degree.indices(self.degree + 1)
-            return [self._vector[degree] for degree in range(*indices)]
+            start, stop, step = degree.indices(self.degree + 1)
```

Comment:
Since we apply the operation directly to the `_vector`, isn't it better to use `degree.indices(len(self._vector))` instead without indirectly asserting that its length is equal to `self.degree + 1`?

Reply:
In that case, we can just access _vector with the slice directly, and avoid the len(self._vector)) call, since the list also implements this behaviour.